updated breadCrumbs and replaced with back va-link for compose flow urls

6StringCode opened the pull request:
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No

## Summary
This pull request introduces significant improvements to the navigation and breadcrumb logic within the secure messaging application's compose flow and newer curated flow, especially around the "Recent Care Teams" and "Select Care Team" steps. The main goals are to provide a more intuitive back navigation experience, ensure the correct entry and exit points for the compose flow, and improve test coverage for these changes.

Key changes include:
**Compose Flow Navigation & Breadcrumb Improvements:**

* Simplified the SmBreadcrumbs component by removing the conditional back button text logic ("Back" vs "Back to [page name]"). The component now always displays a plain "Back" button on pages that use the back link instead of full breadcrumbs.

* Refactored the `SmBreadcrumbs` component to dynamically determine back navigation targets within the compose flow, including handling the new "Recent Care Teams" step and correctly persisting the compose entry point using session storage. This ensures that the "Back" link always leads users to the expected previous step, whether that's the interstitial compose page or the recent care teams list, depending on the user's journey. 

* Updated the logic in `RouteLeavingGuard` to allow navigation to the "Recent Care Teams" path as part of the allowed paths for the compose flow, and normalized path comparisons to avoid issues with trailing slashes.

* Modified navigation in the `RecentCareTeams` container to consistently remove trailing slashes from redirect paths, ensuring uniformity and preventing navigation bugs. 

Team: MHV Secure Messaging team owns this component

## Related issue(s)

This PR fixes issue from two related tickets:
- [https://github.com/department-of-veterans-affairs/va.gov-team/issues/120572](https://github.com/department-of-veterans-affairs/va.gov-team/issues/120572)
- [https://github.com/department-of-veterans-affairs/va.gov-team/issues/120574](https://github.com/department-of-veterans-affairs/va.gov-team/issues/120574)

## Testing done
- Old behavior: 
  - The back button would conditionally show either "Back" or "Back to [folder name]" depending on which page you were on (determined by the backBreadcrumb boolean)
  - Clicking the Back link from the Select Care Page would go to the Interstitial Page instead of directly back to the Recent care page.
- New behavior: 
  - The back button consistently shows "Back" on all pages that use the shortened breadcrumb view, and points to the correct href.
  - When in the Compose Message Flow, the Back link goes back to the previous page in the history of the User flow.

- Steps to verify:
  1. Navigate through the compose message flow. Inbox should show breadcrumbs, but all subsequent pages in the flow should show the Back link which should take you back to your previous page wherever you are in the flow
  3. Test the smart navigation logic by clicking back from different contexts (compose flow, drafts, folders, etc.)
Verify all pages that should show full breadcrumbs still do (inbox, sent, folders main pages)
Results: All navigation flows work as expected. The smart navigateBack() logic remains intact and functional. Full breadcrumbs display correctly on appropriate pages.
  4. Unit tests were updated to test the correct updated URL flow.

 - ## Test Coverage Improvements
**Added comprehensive unit test coverage:**

- **SessionStorage Management** (4 tests): Validates entry URL capture when navigating into compose flow from different folder contexts
- **Folder-Specific Navigation** (3 tests): Tests back navigation logic from Sent, Inbox, and Folders pages  
- **Invalid Folder ID Redirects** (4 tests): Validates system folder ID handling (0, -1, -2) and custom folder paths
- **Custom Folder Breadcrumbs** (1 test): Verifies breadcrumb setup for user-created folders

**Test Results:** All 23 unit tests passing. Coverage increased from baseline to 98.05% statements, 94.31% branches, 96.15% functions.

**Added comprehensive E2E test coverage:**
- Entry Point Preservation (2 tests): Validates users return to their starting folder (Inbox/Sent/Folders) after backing out of compose flow, with sessionStorage verification
- Recent Care Teams Integration (2 tests): Tests complete forward/backward navigation through recent care teams step, including entry point preservation from different folders
- Basic Flow Navigation (2 tests): Validates back button behavior through compose flow steps and care team help page navigation

## What areas of the site does it impact?
Secure Messaging

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.

6StringCode committed:
updated breadCrumbs and replaced with back va-link for compose flow urls

6StringCode committed:
updated full url on breadcrumbs unit test

6StringCode commented on src/applications/mhv-secure-messaging/components/shared/SmBreadcrumbs.jsx:
```diff
@@ -77,8 +65,7 @@ const SmBreadcrumbs = () => {
   }`;
-  const shortenBreadcrumb = pathsWithShortBreadcrumb.includes(crumbPath);
-  const backBreadcrumb = pathsWithBackBreadcrumb.includes(crumbPath);
+  const shortenBreadcrumb = pathsUsingBackLink.includes(crumbPath);
```
Done.

6StringCode committed:
fixed buf preventing select-care-team page to go back to recent. Added unit test coverage

6StringCode committed:
updated previousURL navigation logic so that only recent flow goes back to previous url

6StringCode committed:
Fix breadcrumb back button loops in compose message flow

Replace dynamic previousUrl logic with explicit composeFlowMap defining back button destinations for each wizard page. Add safety checks for undefined/invalid previousUrl values to prevent navigation loops.

Commit:
Normalize slash and remove exit point

6StringCode committed:
combine and update changes to smbreadcrumbs

6StringCode committed:
cleaned up url destinations

6StringCode committed:
fixed breadcrumbs.curated unit test

6StringCode committed:
fixed existing unit tests

6StringCode committed:
Add unit tests for breadcrumb sessionStorage and compose flow navigation

6StringCode committed:
Add breadcrumb navigation unit tests and remove dead sessionStorage code

Covers compose entry URL capture, folder navigation (Sent/Inbox/Folders), and contact list flow.

6StringCode committed:
Add unit tests for system folder ID redirect validation

6StringCode committed:
Revert yarn.lock to match main branch

6StringCode committed:
unit tests fully updated to > 95% coverage

6StringCode committed:
Fix Node 22 compatibility: remove Storage.prototype spying

6StringCode committed:
Merge branch 'main' into 120572-staging-review-finding-breadcrumbs-incorrect

alex-morgun-1 reviewed (commented):
Adam, this is great. Thanks for going an extra mile an fixing those side effects we ran into. Made couple notes. Also, I think this sceanrio with breadcrumbs is one of those examples where e2e testing is crucial. Do you think we can add e2e testing for breadcrumb scenarios on the new curated list flow?

6StringCode committed:
standardized trailing / in paths in constants.js

6StringCode committed:
updated trailing / in e2e constants.js

6StringCode committed:
Add E2E tests for breadcrumb navigation in curated compose flow

Validates entry point preservation, recent care teams integration, and complete navigation cycles.

6StringCode committed:
Add feature flag check to breadcrumb navigation for recent recipients flow

Check mhv_secure_messaging_recent_recipients flag before routing to /recent/ page, preventing navigation loops when feature is disabled. Also refactor currentPath construction for consistency.

6StringCode committed:
- Fix contact list back navigation to use previousUrl directly, avoiding timing issues where fallbackUrl was calculated with stale Redux state

6StringCode committed:
update 2 e2e tests to reflect new back button flow changes

Tonksthebear reviewed (approved):
LGTM. Thanks for tackling this one
